lib/rubocop/cop/util.rb

```diff
@@ -212,32 +212,32 @@ def double_quotes_required?(string)
 
         # Regex matches IF there is a ' or there is a \\ in the string that is
         # not preceded/followed by another \\ (e.g. "\\x34") but not "\\\\".
-        string.inspect =~ /'|(?<! \\) \\{2}* \\ (?![\\"])/x
+        string =~ /'|(?<! \\) \\{2}* \\ (?![\\"])/x
       end
 
       # If double quoted string literals are found in Ruby code, and they are
       # not the preferred style, should they be flagged?
       def double_quotes_acceptable?(string)
         # If a string literal contains hard-to-type characters which would
         # not appear on a "normal" keyboard, then double-quotes are acceptable
-        double_quotes_required?(string) ||
+        needs_escaping?(string) ||
           string.codepoints.any? { |cp| cp < 32 || cp > 126 }
       end
 
+      def needs_escaping?(string)
+        double_quotes_required?(string.inspect)
```

Comment:
> You're right, "escape" makes more sense. Should I extract that method then?

I don't think it's necessary.

> Also, even if the tests pass in both ways, I think string.inspect[1..-2] makes more sense than just string.inspect since inspect adds double quotes around the string that are not relevant for this use case.

I agree.